pkg/provider/azure_storageaccount.go

```diff
@@ -271,9 +273,6 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 			}
 		}
 
-		if location == "" {
```

Comment:
why remove this?

Reply:
move it to [here](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2998/files#diff-43399bb105b7d123e4c10ca28097886143c41c198bdd8189be09eba0a5d70fdaR171) since we need to use it at [here](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2998/files#diff-43399bb105b7d123e4c10ca28097886143c41c198bdd8189be09eba0a5d70fdaR390)